Documentation/troubleshooting.md

```diff
@@ -34,3 +34,17 @@ Account: [myname@example.org]
 $ kubectl create clusterrolebinding myname-cluster-admin-binding --clusterrole=cluster-admin --user=myname@example.org
 Clusterrolebinding "myname-cluster-admin-binding" created
 ```
+
+### Troubleshooting ServiceMonitor changes
+
+When creating/deleting/modifying `ServiceMonitor` objects it is sometimes not as obvious what piece is not working properly. This section gives a step by step guide how to troubleshoot such actions on a `ServiceMonitor` object.
+
+#### Has my `ServiceMonitor` been picked up by Prometheus?
+
+`ServiceMonitor` objects are selected by the `serviceMonitorSelector` of a Prometheus object. The name of a `ServiceMonitor` is encoded in the Prometheus configuration, so you can simply grep whether it is present there. The configuration generated by the Prometheus Operator is stored in a Kubernetes `Secret`, named after the Prometheus object name prefixed with `prometheus-` and is located in the same namespace as the Prometheus object. For example for a Prometheus object called `k8s` one can find out if the `ServiceMonitor` named `my-service-monitor` has been picked up with:
+
+```
+kubectl -n monitoring get secret prometheus-k8s -ojson | jq -r '.data["prometheus.yaml"]' | base64 -d | grep "my-service-monitor"
```

Comment:
Nice command. I always simply go to the Prometheus Configuration UI. But I guess then there's the reload in between?

Reply:
also this accidentally ended up in this PR I actually wanted to create a separate one for the troubleshooting stuff, but is too much work to remove :grin: